### DISCUSSION TITLE
Remove PHPCR layer completely, or at least make it optional

### DISCUSSION THREAD

**Discussion by Jurek-Raben** [sulu/sulu#7305] [Ideas]
Sulu is a great cms with a lot of benefits, and also currently the only headless cms which runs symfony 6.4. 

It uses a PHPCR standard to store any content data... But this standard does not seem to be very handy at all, and the benefit seems to be questionable, too. Also this decision comes with a bunch of disadvantages:

- Content is not searchable and changeable via normal queries, only using a document manager. So php is required in any case.
- Contents even are encapsulated into a huge xml structures. So XML within a table entry. It makes it very hard to debug, to alter data, and also is very, very slow
- PHPCR comes from the JAVA world, where requirements from reality might not count so much. In my opinion, this is a very questionable design.
- Also XML is not easy to maintain and produces an overhead, too
- This complete structure is not human readable at all

So my suggestion would be to make this whole layer optional, and instead use a human readable, most simple database relation structure, which can be easily queried, manipulated from outside and performs well. Maybe there is a common standard for these structures, too, I don't know. Common are a tree table, a category/page table, and content table. Where i18n is required, it should use a join table, to have no redundacy.

Replacing PHPCR and all XML configuration. Instead using lightweight structures like JSON.

**Comment by marcagrio** [comment]
A few years ago there was talk of replacing PHPCR: "[we’ll be replacing our current content storage solution, PHPCR, with our own content storage tool](https://sulu.io/blog/looking-ahead-to-2022-we-are-grateful-to-the-sulu-community)". But it has not happened (yet) and I don't think it was mentioned publicly again.

**Comment by mamazu** [comment]
Yes, in Sulu 3 PHPCR is going to be replaced with the Sulu Content Bundle which is based on JSON.
